tkg/client/upgrade_cluster.go

```diff
@@ -181,18 +181,35 @@
 	return c.DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient, options)
 }
 
-func (c *TkgClient) DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) error {
+func (c *TkgClient) DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) (err error) {
```

Comment:
**suggestion:** rename `err` to `retErr` to avoid accidental assignment to the return variable in the method body. It's best to _only_ access `retErr` in the _deferred_ func.

In the method body, write return statements as usual (_do not_ assign to `retErr`).

Reply:
done